transports/transports_test.go

```diff
@@ -34,6 +34,8 @@ func TestImageNameHandling(t *testing.T) {
 		{"docker", "//busybox:notlatest", "//busybox:notlatest"}, // This also tests handling of multiple ":" characters
 		{"docker-daemon", "sha256:0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef", "sha256:0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"},
 		{"docker-daemon", "busybox:latest", "busybox:latest"},
+		{"docker-archive", "/var/lib/oci/busybox.tar:busybox:latest", "/var/lib/oci/busybox.tar:busybox:latest"},
+		{"docker-archive", "busybox.tar:busybox:latest", "busybox.tar:busybox:latest"},
```

Comment:
(I expect these test cases to fail now; either update them, or perhaps use `FamiliarString` in the `StringWithinTransport` method.)